jwcrypto/jwa.py

```diff
@@ -379,6 +379,23 @@ class _Rsa15(_RSA, JWAAlgorithm):
     def __init__(self):
         super(_Rsa15, self).__init__(padding.PKCS1v15())
 
+    def unwrap(self, key, bitsize, ek, headers):
+        self._check_key(key)
+        # Address MMA attack by implementing RFC 3218 - 2.3.2. Random Filling
+        # provides a random cek that will cause the decryption engine to
+        # run to the end, but will fail decryption later.
+
+        # always generate a random cek so we spend roughly the
+        # same time as in the exception side of the branch
+        cek = _randombits(bitsize)
+        try:
+            cek = super(_Rsa15, self).unwrap(key, bitsize, ek, headers)
+            # always raise so we always run through the exception handling
+            # code in all cases
+            raise Exception('Dummy')
```

Comment:
It does, in my testing the difference between valid CEK and invalid CEKis still measurable in a tight lopp on the local machine, but hardly over the network where the rest of the stack noise is greater than the timing difference. raising an exception there lowered the difference between the code path of about .4% on average in my testing, so I added it. It is not perfect, but can't be,

Note that I measured up to 3% variability in decoding 2 identical messages encrypted with difference CEKs, so as long as we stay in the error margin the attacker has too much noise to figure out what he got.